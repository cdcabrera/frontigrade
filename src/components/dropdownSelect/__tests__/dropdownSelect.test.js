import React from 'react';
import { mount } from 'enzyme';
import { MenuItem } from 'patternfly-react';
import DropdownSelect from '../dropdownSelect';

describe('DropdownSelect Component', () => {
  it('should render', () => {
    const props = {
      id: 'test',
      options: [{ title: 'lorem', value: 'ipsum' }, { title: 'hello', value: 'world' }]
    };

    const component = mount(
      <DropdownSelect {...props}>
        <MenuItem eventKey="ipsum">Lorem ipsum</MenuItem>
      </DropdownSelect>
    );

    expect(component.render()).toMatchSnapshot('basic dropdown');

    component.setProps({
      selectValue: ['world', 'ipsum'],
      multiselect: true
    });

    expect(component.render()).toMatchSnapshot('multiselect dropdown');
  });
});
