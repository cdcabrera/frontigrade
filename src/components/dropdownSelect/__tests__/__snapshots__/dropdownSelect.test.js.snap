// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownSelect Component should render: basic dropdown 1`] = `
<div
  class="cloudmeter-dropdownselect  dropdown btn-group"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="dropdown-toggle btn btn-default"
    id="test"
    role="button"
    type="button"
  >
    <span>
      Select option
    </span>
     
    <span
      class="caret"
    />
  </button>
  <ul
    aria-labelledby="test"
    class="dropdown-menu"
    role="menu"
  >
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        lorem
      </a>
    </li>
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        hello
      </a>
    </li>
  </ul>
</div>
`;

exports[`DropdownSelect Component should render: multiselect dropdown 1`] = `
<div
  class="cloudmeter-dropdownselect  dropdown btn-group"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="dropdown-toggle btn btn-default"
    id="test"
    role="button"
    type="button"
  >
    <span>
      Select option
    </span>
     
    <span
      class="caret"
    />
  </button>
  <ul
    aria-labelledby="test"
    class="dropdown-menu"
    role="menu"
  >
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="cloudmeter-dropdownselect-menuitem row"
        >
          <div
            class="cloudmeter-dropdownselect-menuitemname col-xs-10"
          >
            lorem
          </div>
          <div
            class="cloudmeter-dropdownselect-menuitemcheck col-xs-2"
          />
        </div>
      </a>
    </li>
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="cloudmeter-dropdownselect-menuitem row"
        >
          <div
            class="cloudmeter-dropdownselect-menuitemname col-xs-10"
          >
            hello
          </div>
          <div
            class="cloudmeter-dropdownselect-menuitemcheck col-xs-2"
          />
        </div>
      </a>
    </li>
  </ul>
</div>
`;
